jbmc/src/java_bytecode/java_bytecode_internal_additions.cpp

```diff
@@ -15,8 +15,42 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/cprover_prefix.h>
 #include <util/c_types.h>
 
+#include "java_types.h"
+
 void java_internal_additions(symbol_table_baset &dest)
 {
+  // add monitorenter
+
+  {
+    code_typet type;
+    type.return_type() = void_typet();
+    type.parameters().resize(1);
+    type.parameters()[0].type() = java_reference_type(void_typet());
```

Comment:
Perhaps an Object *, since that defines the @lock field they are expected to use?